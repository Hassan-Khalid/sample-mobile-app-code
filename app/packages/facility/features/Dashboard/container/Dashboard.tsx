import React from 'react';
import {
  View, Text, Button, SafeAreaView,
} from 'react-native';
import useDashboard from '../hooks/useDashboard';

function Dashboard() {
  const { navigateToLogin } = useDashboard();
  return (
    <SafeAreaView style={{ flex: 1 }}>
      <View>
        <Text>Welcome to Facility Dashboard</Text>
        <Button title="login" onPress={navigateToLogin} />
      </View>
    </SafeAreaView>
  );
}
export default Dashboard;
