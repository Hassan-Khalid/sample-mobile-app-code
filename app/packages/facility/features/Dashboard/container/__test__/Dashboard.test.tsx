import React from 'react';
import { render } from '@testing-library/react-native';
import Dashboard from '../Dashboard';

describe('Dashboard screen', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render container', async () => {
    // exercise
    const wrapper = render(<Dashboard />);

    // post-conditions
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
