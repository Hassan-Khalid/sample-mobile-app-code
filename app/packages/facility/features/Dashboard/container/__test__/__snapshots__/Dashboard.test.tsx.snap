// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard screen should render container 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <View>
    <Text>
      Welcome to Facility Dashboard
    </Text>
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          login
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
